src/cs50/sql.py

```diff
@@ -125,6 +130,12 @@ def execute(self, sql, *args, **kwargs):
             if token.ttype in [sqlparse.tokens.Keyword.DDL, sqlparse.tokens.Keyword.DML]:
                 command = token.value.upper()
                 break
+
+            # Begin a new session, if transaction started by caller (not using autocommit)
+            elif token.value.upper() in ["BEGIN", "START"]:
```

Comment:
What `ttype` does sqlparse assign to BEGIN and START? We should probably check for that too, for good measure?

Reply:
sqlparse is inconsistent here, part of why I didn't check against it in the first place -- `BEGIN` is just a `Keyword`, no further classification; `START` is `Keyword.DML`.

Reply:
Ugh, just for good measure, mind seeing if there's another parser we should just use, which might also help with https://github.com/cs50/python-cs50/issues/117?

Reply:
sqlparse is probably the best one for our needs -- after many pages of Google, there really doesn't seem to be anything else with the features we need available right now. The closest is this [project from Uber](https://github.com/uber/queryparser) but it doesn't currently support any of SQLite, MySQL, or PostgreSQL and hasn't been updated in over a year. [Other](https://github.com/mastermay/sql-parser) [options](https://github.com/TwoLaid/python-sqlparser) don't implement key features (like UPDATE/INSERT statements!), similarly haven't been updated in years, and sometimes aren't cross-platform.